### PR TITLE
Introduce empty targets to ease the internal merge of #25586 

### PR DESCRIPTION
**Pull request by yihuazhang**
This PR is a subset of #25586. When performing internal patches, we noticed it touches a large number of internal build targets so that they can depend on the new targets introduced in #25586. The downside is that it also involves a large number of reviewers which will make the rollback process slow and difficult. In order to ease the process, in this PR, we introduced new empty targets that will be depended by internal targets, which will be sufficient to make internal changes via LSC. 

Note that this PR includes the minimum code changes needed to successfully build the new targets.

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -3341,6 +3341,266 @@ grpc_cc_library(
...
+    external_deps = [
+        "libssl",
+    ],
+    language = "c++",
+)
+
+grpc_cc_library(
+    name = "tsi_base",
+    srcs = [
+        "src/core/tsi/transport_security.cc",
```
Comment (markdroth) [inline BUILD:3358]: Is it going to cause problems that we have the same files listed in multiple targets here?

Could we avoid this by removing these files from their current targets and having those targets depend on these new targets?  For example, in the case of this `tsi_base` target, we could do the following:
- Remove `src/core/tsi/transport_security.cc`, `src/core/tsi/transport_security.h`, and `src/core/tsi/transport_security_interface.h` from the existing `tsi_interface` target, and have `tsi_interface` depend on `tsi_base`.
- Remove `src/core/tsi/transport_security_grpc.cc` and `src/core/tsi/transport_security_grpc.h` from the `tsi` target, and have `tsi` depend on `tsi_base`.

Similar things can be done for the other targets.

Comment (yihuazhang) [reply BUILD:3358]: Do you mean causing problems internally or in OSS? Internally (cl/407803147), we will replace the dependency of `grpc_secure` with `grpc_security_base`  and `tsi_interface` with `tsi_base`, respectively which means the internal targets will not depend on two targets that include the same files. WDYT?

Comment (yihuazhang) [reply BUILD:3358]: Sorry, let me take back my previous comment. It indeed results in a build failure internally due to the same file inclusion in multiple targets. Let me fix it as you suggested.

**Comment by yihuazhang** [comment]
Hi @markdroth, the PR is ready for review again :)

**Review by markdroth — APPROVED** [review]
This looks great!

**Comment by yihuazhang** [comment]
https://github.com/grpc/grpc/issues/25856,

**Pull request merged**